service/controller/v2/resource/deployment/resource.go

```diff
@@ -97,16 +99,38 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			},
 		},
 	}
-	f, err := deploymentsClient.CreateOrUpdate(ctx, resourceGroupName, mainDeployment.Name, d)
-	if err != nil {
+
+	d, err := deploymentsClient.Get(ctx, resourceGroupName, mainDeployment.Name)
+	if IsNotFound(err) {
+		_, err := deploymentsClient.CreateOrUpdate(ctx, resourceGroupName, mainDeployment.Name, newDeployment)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring deployment in progress")
+		reconciliationcanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation for custom object")
+
+		return nil
```

Comment:
So again, do we need that part when other resources are ready for deployment resource to be async?

Reply:
Uncertain. I cannot decide either way. Do you have an argument or use case or benefit for one way or the other?

Reply:
My argument is simplification. Also if we run into timing issues we spot them earlier.

Reply:
I will check on this.

Reply:
I removed this. Works.